src/sign_tx.c

```diff
@@ -0,0 +1,506 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/*
+ * The sign tx command is a bit more complex than the others, mostly
+ * due to the transaction parsing. It requires multiple iterations from
+ * the wallet to complete and we use p1 values to track this communication.
+ *
+ * On the first packet received, it has the change output info, if one exists.
+ * If there's no change output, it's just the byte 0x00. Otherwise, it expects
+ * any non 0x00 value, followed by the change output index (1 byte) and the key
+ * index (4 bytes) that the change is supposed to be sent to. This information
+ * is used to confirmed that the change is indeed being sent to an address from
+ * this wallet. Eg: 
+ *      [0x01, 0x03, 0x00, 0x00, 0x00, 0x05]
```

Comment:
I liked the idea of getting the change output from the wallet. But it may be dangerous because a malicious software may tamper the communication between the computer and the ledger, changing the index of the change output to hide his output.

One way to go is to validate that the change output belongs to this wallet. If it doesn't, we can show a warning message or simply decline the transaction. What do you think?

Reply:
Yes, this is already done. We check if the change output belongs to the wallet. Take a look at `verify_change_output`. If it fails, we send an error back to the wallet.

Reply:
Great! Thanks! Can you just update the description to include this validation, please?

Reply:
Done